packages/xo-server/src/api/vm.js

```diff
@@ -756,6 +757,27 @@ export { convertToTemplate as convert }
 
 // -------------------------------------------------------------------
 
+export async function cloneToTemplate({ vm }) {
+  const xapi = await this.getXapi(vm)
+  const clonedVm = await xapi.copyVm(vm._xapiId, {
+    nameLabel: vm.name_label,
+  })
+  try {
+    await clonedVm.set_is_a_template(true)
+  } catch (e) {
+    await xapi.VM_destroy(clonedVm.$ref)
```

Comment:
To not swallow the source error,  I suggest:

```suggestion
    ignoreErrors.call(xapi.VM_destroy(clonedVm.$ref))
    throw error
```